client/app/lib/providers/computecontroller.coffee

```diff
@@ -1203,22 +1213,56 @@ module.exports = class ComputeController extends KDController
       kd.singletons.router.handleRoute route
 
 
+  shareStackWithTeam: (stackTemplate, callback) ->
+
+    { reactor, groupsController } = kd.singletons
+    { credentials, config: { requiredProviders } } = stackTemplate
+
+    createShareModal (needShare, modal) =>
+
+      groupsController.setDefaultTemplate stackTemplate, 'share', (err) =>
+        if needShare
+        then @shareCredentials credentials, requiredProviders, -> modal.destroy()
+        else modal.destroy()
+
+
+  makeStackPrivate: (stackTemplate) ->
```

Comment:
this shouldn't have a UI code here, just the functionality should be here. If you need to provide some UI helper you can put them into `ComputController.UI`